test/sql/collate/collate_union.test

```diff
@@ -0,0 +1,130 @@
+# name: test/sql/collate/collate_union.test
+# description: Test collation in Union operator
+# group: [collate]
+
+require icu
+
+query I
+SELECT 'A' COLLATE NOCASE UNION SELECT 'a'
```

Comment:
Can we add some more tests?

* `EXCEPT` and `INTERSECT` tests
* Tests using `default_collation` instead of `COLLATE` expressions
* Tests using collations stored in columns
* Tests using mismatching collations
* Tests using collations in nested set operations (e.g. `(SELECT 'A' UNION SELECT 'b') UNION (SELECT 'c' UNION SELECT 'A' COLLATE NOCASE);`)
* Tests with ICU collations